extensions/trace_propagators/src/main/java/io/opentelemetry/extensions/trace/propagation/StackTracePropagator.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.extensions.trace.propagation;
+
+import io.grpc.Context;
+import io.opentelemetry.context.propagation.HttpTextFormat;
+import io.opentelemetry.trace.TracingContextUtils;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * A propagator designed to inject and extract multiple trace {@code HttpTextFormat} propagators,
+ * intendended for backwards compatibility with existing services using different formats. It works
+ * in a stack-fashion, starting with the last registered propagator, to the first one.
+ *
+ * <p>Upon injection, this propagator invokes {@code HttpTextFormat#inject()} for every registered
+ * trace propagator. This will result in the carrier containing all the registered formats.
+ *
+ * <p>Upon extraction, this propagator invokes {@code HttpTextFormat#extract()} for every registered
+ * trace propagator, returning immediately when a successful extraction happened.
+ *
+ * <pre>{@code
+ * HttpTextFormat traceFormats = StackTracePropagator.builder()
+ *   .addPropagator(new MyCustomTracePropagator())
+ *   .addPropagator(new JaegerPropagator())
+ *   .addPropagator(new HttpTraceContext())
+ *   .build();
+ * // Register it in the global propagators:
+ * OpenTelemetry.setPropagators(
+ *     DefaultContextPropagators.builder()
+ *       .addHttpTextFormat(traceFormats)
+ *       .build());
+ * ...
+ * // Extraction will be performed in reverse  order, i.e. starting with the last
+ * // registered propagator (HttpTraceContext in this example).
+ * Context context = OpenTelemetry.getPropagators().getHttpTextFormat()
+ *   .extract(context, carrier, carrierGetter);
+ * }</pre>
+ *
+ * @since 0.6.0
+ */
+@Immutable
+public class StackTracePropagator implements HttpTextFormat {
```

Comment:
I think calling this stack trace will confuse many people. How about `StackedTracePropagator`, or `TraceMultiPropagator` or `CompoundPropagator`?

Reply:
Yup :P `Composite` is pretty common I think too. Either way, I think the options open up if we don't go in reverse order, which from what I can tell doesn't seem to be that important.

Reply:
Ah also remembered we have multispanprocessor. So `MultiPropagator` would be consistent for this codebase.

Reply:
`CompoundPropagator` is too generic. `StackedTracePropagator` sounds great to me if we keep the stack-alike behavior, but `TraceMultiPropagator` would work if we go the simple-linear behavior.

`MultiPropagator` is not a good name IMHO as we don't *really* work on all propagators for extraction (we stop as soon as one properly extracts a `SpanContext`).